# OMetaNet
Accurate identification of RNA 2'-O-methylation (2OM) sites is crucial for understanding RNA regulatory mechanisms, yet existing prediction tools suffer from limited accuracy and insufficient capture of sequence-site associations. This study proposes a deep learning framework, OMetaNet, featuring key innovations across three levels: data, features, and model architecture. At the data level, we integrate multiple public sources and apply rigorous cleaning to construct a high-quality, low-redundancy, and class-imbalanced dataset. At the feature level, we design a novel KN-PairMatrix encoding scheme that incorporates physicochemical properties of RNA to generate multidimensional embeddings. At the model level, we develop a feature integration module combining channel attention and cross-modal interaction, introduce a contrastive learning–driven adaptive hybrid loss function, and implement a progressive feature decoupling strategy to enhance the learning of 2OM-specific patterns. Independent evaluations demonstrate that OMetaNet significantly outperforms existing methods in predicting 2OM sites across all four nucleotide types. This study provides an efficient predictive tool for epitranscriptomic research, with the potential to reshape transcriptomic analysis paradigms and offer valuable applications in biomarker discovery and nucleic acid drug development.

Due to the size limit of GitHub upload file, we have uploaded embedding and weight to onedrive:
dataset：https://1drv.ms/f/c/6c930042add3d335/EoBWAoCt8MFEuRvhvoJOd_0B3mPKk-2kZRhaICM6tXsOzg?e=wqMcvP

